File: backend/app/dto/comment.py
from pydantic import BaseModel
from datetime import datetime as datetime

class CommentCreationDTO(BaseModel):
    review_id: int
    text: str

class CommentGetDTO(BaseModel):
    id: int
    review_id: int
    text: str
    likes: int
    liked_by_user: bool = False
    created_at: datetime
    user_id: int
    is_deletable: bool
    user_name: str

File: frontend/app-front/services/api/flixy/types/comment.ts
export interface CommentCreation {
  review_id: number;
  text: string;
}

export interface CommentDataGet {
  id: number;
  user_id: number;
  is_deletable: boolean;
  review_id: number;
  text: string;
  created_at: Date;
  likes: number;
  user_name: string;
  liked_by_user: boolean;
}
